File: QRCode/ViewController.m
//
//  ViewController.m
//  QRCode
//
//  Created by rc on 15/9/24.
//  Copyright © 2015年 rc. All rights reserved.
//

#import "ViewController.h"
#import "QRCode.h"
@interface ViewController ()
@property (nonatomic,strong) UIImageView *imageView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    _imageView = [[UIImageView alloc] initWithFrame:CGRectMake(10, 100,300 , 300)];
    
    _imageView.image = ({
    
        QRCode *qrcode = [[QRCode alloc] init];
        qrcode.dataString = @"www.baidu.com";
        qrcode.size = _imageView.bounds.size;
        qrcode.errorCorrection = ErrorCorrectionHigh;
        qrcode.frontColor = [UIColor redColor];
        qrcode.backgroudColor = [UIColor cyanColor];
        qrcode.iconImage  = [UIImage imageNamed:@"dog.jpg"];
        qrcode.iconRadius = 8;
        qrcode.iconBorderWidth = 10;
        qrcode.iconBorderColor = [UIColor whiteColor];
        [qrcode qrImage];
    
    });
    
    [self.view addSubview:_imageView];
    
    UISlider *slider = [[UISlider alloc] initWithFrame:CGRectMake(20, 420, self.view.bounds.size.width-20*2, 20)];
    slider.value = 1;
    [slider addTarget:self action:@selector(changeImageViewScale:) forControlEvents:UIControlEventValueChanged];
    [self.view addSubview:slider];
    
    
    
    UIButton *saveToAlbumBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    [saveToAlbumBtn setTitleColor:[UIColor redColor] forState:UIControlStateNormal];
    [saveToAlbumBtn setFrame:CGRectMake(0, 450, self.view.frame.size.width, 40)];
    [saveToAlbumBtn setTitle:@"保存到相册" forState:UIControlStateNormal];
    [saveToAlbumBtn addTarget:self action:@selector(saveToAlbum) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:saveToAlbumBtn];
}

- (void)changeImageViewScale:(UISlider *)slider {
    _imageView.transform = CGAffineTransformMakeScale(slider.value, slider.value);
}

- (void)saveToAlbum {

    UIImageWriteToSavedPhotosAlbum(_imageView.image, self, nil, NULL);
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: QRCodeWithSwift/ViewController.swift
//
//  ViewController.swift
//  QRCodeWithSwift
//
//  Created by tassel on 2019/6/30.
//  Copyright © 2019 rc. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var imageView:UIImageView?
    
    @objc func changeImageViewScale(sender:UISlider){
        
        imageView?.transform = CGAffineTransform(scaleX: CGFloat(sender.value), y: CGFloat(sender.value))
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        imageView = UIImageView(frame: CGRect(x: 10, y: 100, width: 300, height: 300))
        
        imageView!.image = {
            var qrCode = QRCode("http://www.baidu.com")!
            qrCode.errorCorrection = .High
            qrCode.size = (imageView?.frame.size)!
            qrCode.color = CIColor(rgba: "f1466a")
            qrCode.backgroundColor = CIColor(rgba: "ffffff")
            qrCode.iconImage = UIImage(named: "avatar")
            qrCode.iconBorderWidth = 8
            qrCode.iconBorderColor = UIColor.white
            qrCode.iconRadius = 10
            
            return qrCode.image
        }()
        
        self.view.addSubview(imageView!)
        
        let slider = UISlider(frame: CGRect(x: 20, y: 420, width: self.view.bounds.size.width-20*2, height: 20))
        slider.value = 1
        slider.addTarget(self, action: #selector(changeImageViewScale(sender:)), for: .valueChanged)
        self.view.addSubview(slider)
        
    }
  
    
    
}

